src/test/scala/is/hail/expr/ir/MathFunctionsSuite.scala

```diff
@@ -94,4 +97,59 @@ class MathFunctionsSuite extends TestNGSuite {
       Uniroot("x", Ref("x", tfloat), NA(tfloat), F64(0)),
       null)
   }
+
+  @Test def chisq() {
```

Comment:
The other tests have been using the `@DataProvider` annotation to do this kind of parameterized testing. Do you want to do that here as well?

Reply:
Is this what you had in mind? Do you prefer it?
```
  @DataProvider(name = "chisq")
  def chisqData(): Array[Array[Any]] = Array(
    Array(0, 0, 0, 0, Double.NaN, Double.NaN),
    Array(0, 1, 1, 1, 0.38647623077123266, 0.0),
    Array(1, 0, 1, 1, 0.38647623077123266, Double.PositiveInfinity),
    Array(1, 1, 0, 1, 0.38647623077123266, Double.PositiveInfinity),
    Array(1, 1, 1, 0, 0.38647623077123266, 0.0),
    Array(10, 10, 10, 10, 1.0, 1.0),
    Array(51, 43, 22, 92, 1.462626e-7, (51.0 * 92) / (22 * 43))
  )
  
  @Test(dataProvider = "chisq")
  def chisq(a: Int, b: Int, c: Int, d: Int, pValue: Double, oddsRatio: Double) {
      val r = eval(invoke("chisq", a, b, c, d)).asInstanceOf[Row]
      assert(D0_==(pValue, r.getDouble(0)))
      assert(D0_==(oddsRatio, r.getDouble(1)))
  }
```

Reply:
yeah. I think the consistency between tests is nice, but I don't have strong feelings outside of that.